src/js/index.js

```diff
@@ -0,0 +1,115 @@
+import tasks from './tasks.js';
+import taskManager from './taskManager.js';
+
+const markupAllLists = () => {
+  let allLists = '';
+  tasks.todoData.forEach((data, index) => {
+    allLists += `<li class="todo-item">
+                    <div class="todo-entry" id=entry_${index}>
+                      <input type="checkbox" ${data.completed ? 'checked' : ''} class="status" title="Check!">
+                      <input type="text" class="todo" value="${data.description}">
+                      <i class="fa-solid fa-trash-can delete hide" title="Delete entry"></i>
+                      <i class="fa-solid fa-ellipsis-vertical reorder" title="Move"></i>
+                    </div>
+                    <hr class="divide">
+                  </li>`;
+  });
+  return allLists;
+};
+
+const renderLists = () => {
+  const todoCollection = document.getElementById('todo-collection');
+  todoCollection.innerHTML = markupAllLists();
+};
+
+renderLists();
+
+// General declaration that re-rendering lists will not affect
+const enterTodo = document.getElementById('enter-todo');
+const enterBu = document.querySelector('.enter');
+const clearCompleted = document.getElementById('clear-completed');
+const refreshBu = document.getElementById('refresh');
+
+const addRefreshingListeners = () => {
+  const descInput = document.querySelectorAll('.todo');
+  const deleteButtons = document.querySelectorAll('.delete');
+  const statuses = document.querySelectorAll('.status');
+
+  descInput.forEach((element) => {
+    ['focus', 'blur', 'keyup'].forEach((evt) => {
+      element.addEventListener(evt, (e) => {
+        const todoEntry = e.target.parentNode;
+
+        // Modify the looks
+        todoEntry.classList.toggle('highlight-input');
+        todoEntry.querySelector('.delete').classList.toggle('hide');
+        todoEntry.querySelector('.reorder').classList.toggle('hide');
+
+        if (evt === 'blur' || (evt === 'keyup' && e.key === 'Enter')) {
+          // It's a modify
+          const taskIndex = Number(e.target.parentNode.id.split('_')[1]);
+          taskManager.modifyTask({ description: e.target.value }, taskIndex);
+          todoEntry.classList.remove('highlight-input');
+          todoEntry.querySelector('.delete').classList.add('hide');
+          todoEntry.querySelector('.reorder').classList.remove('hide');
+          e.target.blur();
+        }
+      });
+    });
+  });
+
+  deleteButtons.forEach((deleteButton) => {
+    deleteButton.addEventListener('mousedown', (e) => {
+      const taskIndex = Number(e.target.parentNode.id.split('_')[1]);
+      taskManager.deleteTask(taskIndex);
+      renderLists();
+      addRefreshingListeners();
+    });
+  });
+
+  statuses.forEach((status) => {
+    status.addEventListener('change', (e) => {
+      const taskIndex = Number(e.target.parentNode.id.split('_')[1]);
+      taskManager.modifyTask({ completed: e.target.checked }, taskIndex);
+    });
+  });
+};
+
+addRefreshingListeners();
+
```

Comment:
-Instead of adding event listeners to each individual element, use event delegation to handle events at a higher level. This reduces the amount of code and simplifies the logic.

```javascript
const addEventDelegationListeners = () => {
  // add event listener to the todo collection to handle events on todo items
  const todoCollection = document.getElementById('todo-collection');
  todoCollection.addEventListener('focus', (e) => {
    const todoEntry = e.target.parentNode;
    todoEntry.classList.add('highlight-input');
    todoEntry.querySelector('.delete').classList.remove('hide');
    todoEntry.querySelector('.reorder').classList.remove('hide');
  }, true);

  todoCollection.addEventListener('blur', (e) => {
    const todoEntry = e.target.parentNode;
    todoEntry.classList.remove
```